homeassistant/components/pvpc_hourly_pricing/config_flow.py

```diff
@@ -33,36 +60,180 @@ async def async_step_user(
         if user_input is not None:
             await self.async_set_unique_id(user_input[ATTR_TARIFF])
             self._abort_if_unique_id_configured()
-            return self.async_create_entry(title=user_input[CONF_NAME], data=user_input)
+            if not user_input[CONF_USE_API_TOKEN]:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
+                    data={
+                        CONF_NAME: user_input[CONF_NAME],
+                        ATTR_TARIFF: user_input[ATTR_TARIFF],
+                        ATTR_POWER: user_input[ATTR_POWER],
+                        ATTR_POWER_P3: user_input[ATTR_POWER_P3],
+                        CONF_API_TOKEN: None,
+                    },
+                )
+
+            self._name = user_input[CONF_NAME]
+            self._tariff = user_input[ATTR_TARIFF]
+            self._power = user_input[ATTR_POWER]
+            self._power_p3 = user_input[ATTR_POWER_P3]
+            self._use_api_token = user_input[CONF_USE_API_TOKEN]
+            return self.async_show_form(
+                step_id="api_token",
```

Comment:
Don't move directly to the form of another step. Await the step method instead. The form is specific for each step method and should only be called from its step method.

Reply:
Hi Martin, thanks for the double-check!

Do you meant this change in #104412, right?

Reply:
Yes.